frontend/packages/kubevirt-plugin/src/components/vms/vm-details.tsx

```diff
@@ -83,13 +82,13 @@ export const VMDetails: React.FC<VMDetailsProps> = (props) => {
   const vmiLike = kindObj === VirtualMachineModel ? vm : vmi;
   const vmServicesData = getServicesForVmi(getLoadedData(props.services, []), vmi);
   const canUpdate = useAccessReview(asAccessReview(kindObj, vmiLike || {}, 'patch')) && !!vmiLike;
-  const [guestAgentInfo] = useGuestAgentInfo({ vmi });
-  const guestAgentInfoOS = getGuestAgentInfoOS(guestAgentInfo);
+
+  const [guestAgentInfoRaw] = useGuestAgentInfo({ vmi });
+  const guestAgentInfo = new GuestAgentInfoWrapper(guestAgentInfoRaw);
+  const operatingSystemID = guestAgentInfo.getOSInfo().getId();
 
   const OSMismatchExists =
-    vmi &&
-    guestAgentInfo &&
-    isWindows(vmiLike) !== (getGuestOSID(guestAgentInfoOS) === 'mswindows');
+    vmi && guestAgentInfoRaw && isWindows(vmiLike) !== (operatingSystemID === 'mswindows');
```

Comment:
We can leave this here for now, but can add it to the `GuestAgentOSInfoWrapper` if you want.